pkg/core/blockchain_test.go

```diff
@@ -2003,3 +2005,258 @@ func setSigner(tx *transaction.Transaction, h util.Uint160) {
 		Scopes:  transaction.Global,
 	}}
 }
+
+func TestBlockchain_StartFromExistingDB(t *testing.T) {
+	newPS := newLevelDBForTestingWithPath
+	ps, path := newPS(t, "")
+	var cfg *config.Config
+	customConfig := func(c *config.Config) {
+		c.ProtocolConfiguration.StateRootInHeader = true // Need for P2PStateExchangeExtensions check.
+		cfg = c
+	}
+	bc := initTestChain(t, ps, customConfig)
+	go bc.Run()
+	initBasicChain(t, bc)
+	require.True(t, bc.BlockHeight() > 5) // Ensure that basic chain is correctly initialised.
+
+	// Information for further tests.
+	h := bc.BlockHeight()
+	cryptoLibHash, err := bc.GetNativeContractScriptHash(nativenames.CryptoLib)
+	require.NoError(t, err)
+	cryptoLibState := bc.GetContractState(cryptoLibHash)
+	require.NotNil(t, cryptoLibState)
+	var (
+		managementID             = -1
+		managementContractPrefix = 8
+	)
+
+	bc.Close() // Ensure persist is done and persistent store is properly closed.
+
+	t.Run("mismatch storage version", func(t *testing.T) {
+		ps, _ = newPS(t, path)
+		t.Cleanup(func() { require.NoError(t, ps.Close()) })
```

Comment:
We always perform this cleanup in this case and in other tests, move it inside the function?